file_browser/src/classes/Folder.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  observable,
+  action,
+  ObservableMap,
+  makeObservable,
+  computed,
+} from "mobx";
+import { SystemItem } from "./SystemItem";
+import fileSystemStore from "../stores/FileSystemStore";
+
+export interface IFolder {
+  id: number;
+  name: string;
+  files: ObservableMap<number, SystemItem>;
+  parent: Folder | null;
+}
+
+export class Folder extends SystemItem implements IFolder {
+  files = observable.map<number, SystemItem>();
+
+  constructor(name: string, parent: Folder | null) {
+    super(name, parent);
+    makeObservable(this, {
+      files: observable,
+      add: action,
+      delete: action,
+      getFilteredMap: computed,
+    });
+  }
+
+  add(item: SystemItem) {
+    this.files.set(item.id, item);
+  }
+
+  delete() {
+    this.files.clear();
+    this.parent?.files.delete(this.id);
+  }
+
+  get getFilteredMap(): ObservableMap<number, SystemItem> {
+    if (fileSystemStore.searchText !== "") {
+      return this.getFilteredMapAssist();
+    }
+    return this.files;
+  }
+
+  private getFilteredMapAssist(): ObservableMap<number, SystemItem> {
```

Comment:
`Assist` is not a word commonly used in programming. Rather, the word `Helper` would be more appropriate here.
But considering the fact that it's only used in one place, it's comploetely pointless to have this method, and the code should move directly into the getter.